pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java

```diff
@@ -392,7 +392,8 @@ public void addFakeServerInstancesToAutoJoinHelixCluster(int numInstances, boole
   public void addFakeServerInstancesToAutoJoinHelixCluster(int numInstances, boolean isSingleTenant, int baseAdminPort)
       throws Exception {
     for (int i = 0; i < numInstances; i++) {
-      addFakeServerInstanceToAutoJoinHelixCluster(SERVER_INSTANCE_ID_PREFIX + i, isSingleTenant, baseAdminPort + i);
+      addFakeServerInstanceToAutoJoinHelixCluster(SERVER_INSTANCE_ID_PREFIX + i, isSingleTenant,
+          NetUtils.findOpenPort(baseAdminPort + i));
```

Comment:
nit: if one port is unavailable netUtils.findOpenPort will increase by 1 --> which essentially makes the next instance i conflict and then tries to increase that as well. should we do truly randomize? what's limiting us to keep using the base + i format?

Reply:
This method finds the open port then create the fake instance, so it won't happen.

Reply:
This method finds the open port then create the fake instance, so it won't happen.

Reply:
This method finds the open port then create the fake instance, so it won't happen.

Reply:
that's even worse right? if one admin port is available and one fake server took it (but not open the port); the next server will took the +1 port as well b/c it is still available --> that means 2 fake server will have the same admin port ?

Reply:
Make those ports purely randomized.